dcos/mesos.py

```diff
@@ -754,7 +766,15 @@ def dict(self):
         :rtype: dict
         """
 
-        return self._task
+        try:
+            fd = self.fault_domain()
+        except Exception:
+            return self._task
+        if not fd:
+            return self._task
```

Comment:
Why make a copy instead of updating the underlying dictionary? And pull the info directly from `self.slave().fault_domain()` (with the `fault_domain()` function for this class just returning `self["domain"]`.

Reply:
my idea was to no mutate the original `self._task` object. However when the user calls `.dict()` a new copy is being made with additional `domain` field injected. assuming the original `self._task` object might be used by other functions etc. Let me know if you think this does not make sense and we better do it in a different way.

Reply:
I think the copy should happen in the `__init__` function and then we are free to modify it ourselves however we please from within this class.

Reply:
I couldn't find places where Task objects are instanciated other than the Framework class in this file, which is also used by the Master class.

I think it's safe to mutate the dict by adding this new key for fault domains.

Reply:
I'm fine with @klueska suggestion too.